src/plugins/embeddable/public/lib/panel/panel_header/panel_header.tsx

```diff
@@ -212,22 +212,24 @@ export function PanelHeader({
   };
 
   return (
-    <figcaption
-      className={classes}
-      data-test-subj={`embeddablePanelHeading-${(title || '').replace(/\s/g, '')}`}
-    >
-      <h2 data-test-subj="dashboardPanelTitle" className="embPanel__title embPanel__dragger">
-        <EuiScreenReaderOnly>{getAriaLabel()}</EuiScreenReaderOnly>
-        {renderTitle()}
-        {renderBadges(badges, embeddable)}
-      </h2>
-      {renderNotifications(notifications, embeddable)}
-      <PanelOptionsMenu
-        isViewMode={isViewMode}
-        getActionContextMenuPanel={getActionContextMenuPanel}
-        closeContextMenu={closeContextMenu}
-        title={title}
-      />
-    </figcaption>
+    <span data-test-subj="dashboardPanelTitle__wrapper">
```

Comment:
If you have the `data-test-subj` above on line 156, do you need to wrap all of this with the same `data-test-subj`?

Reply:
Unfortunately, yup! The `data-test-subj` from line 156 is for when the panel title is explicitly hidden using the toggle - the one on line 215, on the other hand, is for when the title is anything else (including when it is set to the empty string). This is the only way to be able to capture *all* titles regardless of whether or not they are explicitly hidden.